### PR TITLE
adding awicm3 v3.1.1

### PR DESCRIPTION
**Pull request by JanStreffing**
This version of AWI-CM3 is almost identical with v3.1. The only differences are made to be able to run the model with debug flags in OpenIFS.

### CODE REVIEW

**Comment by JanStreffing** [comment]
Set as draft for now. @mandresm and I want to work on a better way to make minor model releases, without the need to copy paste every single major model version default setting.

**Comment by mandresm** [comment]
Hi Jan,

I'm afraid for `esm_master` there is no better way as it doesn't use the `esm_parser` but hard-coded search for specific `choose_` blocks (`choose_version` and `choose_computer.name`). (One day we will fix the esm_master...)

For `esm_runscripts`:

awicm3.yaml
```yaml
general:
    choose_version:
        3.1:
            minor_version: 3.1
        3.1.1:
            minor_version: 3.1
        3.2.1:
            minor_version: 3.2

oifs:
    choose_general.minor_version:
        "3.1":
            var_and_values: for 3.1
        "3.2":
            var_and_values: for 3.2
```

**Comment by mandresm** [comment]
Another option to remove lines in here:
```yaml
general:
    choose_version:
        3.1:
            minor_version: 3.1
        3.1.1:
            minor_version: 3.1
        3.2.1:
            minor_version: 3.2
```

would be:

```yaml
general:
    minor_version: "$(( '.'.join((${version}).split('.')[0:2]) ))"
```

File: configs/components/oifs/oifs.yaml
Comment (pgierz) [inline configs/components/oifs/oifs.yaml:96]: 👍🏻

File: configs/components/oifs/oifs.yaml
Comment (pgierz) [inline configs/components/oifs/oifs.yaml:151]: is this different from other versions (in particular the `comp_command`)? If no, we should just make it universal IMHO

File: configs/couplings/fesom-2.0-awicm-3.1+oifs-43r3-awicm-3.1.1+xios-2.5/fesom-2.0-awicm-3.1+oifs-43r3-awicm-3.1.1+xios-2.5.yaml
Comment (pgierz) [inline configs/couplings/fesom-2.0-awicm-3.1+oifs-43r3-awicm-3.1.1+xios-2.5/fesom-2.0-awicm-3.1+oifs-43r3-awicm-3.1.1+xios-2.5.yaml:10]: Mmmmpf.....changing fortran via sed gives me a small headache. But we also change makefiles via sed.

So, at the end: hey, if it works, hooray 😄

**Review by pgierz — COMMENTED** [review]
Looks good @JanStreffing! Thumbs up from my side 👍🏻

**Comment by pgierz** [comment]
> One day we will fix the esm_master...

Yes, one day I will go crazy and rewrite it all in Rust, and then no one will know how it works anymore even remotely ;-)

**Comment by JanStreffing** [comment]
dont merge yet please. i have not done a test simulation. I suspect it wont work, as some of the blocks the blocks that are choose v3.1 (and there are quite a number of them) would not trigger.

**Comment by pgierz** [comment]
> dont merge yet please. i have not done a test simulation. I suspect it wont work, as some of the blocks the blocks that are choose v3.1 (and there are quite a number of them) would not trigger.

Didn't plan on merging this yet ;-) Just wanted to give you some preliminary feedback

File: configs/components/oifs/oifs.yaml
Comment (JanStreffing) [reply configs/components/oifs/oifs.yaml:151]: the comp command has changed in the past, e.g. with XIOS integration. It will change further e.g. with single precision.

**Comment by JanStreffing** [comment]
From my side ready. Failed tests were all AWI-CM3 and related to the changes made.

**Comment by mandresm** [comment]
I'd like to review this tomorrow, before it's merged.

**Review by mandresm — APPROVED** [review]
Working on the last steps before merging...

**Comment by mandresm** [comment]
I just transferred the esm_tests from 3.1 to 3.1.1 and ran them on Levante. All working nicely :)

**Comment by mandresm** [comment]
#approve-changes

**Comment by JanStreffing** [comment]
> I just transferred the esm_tests from 3.1 to 3.1.1 and ran them on Levante. All working nicely :)

How much work is that? Do you plan to do that for each minor release? I have 3.1.2 and 3.1.3 already coming up. I don't want you to spend all day moving the tests around :D

**Comment by mandresm** [comment]
It's pretty straight forward: changing the version on the test runscript, moving 2 folders to the new name and updating one yaml. If we are going to be doing this often I'll make a small python script so that I don't forget and I can do it with just one command.

**Comment by mandresm** [comment]
But yes, I'll do it next time only with 3.1.3. I think we should only have in the testing all the major versions, 3.1, 3.2, .... in their latest minor state.

**Comment by mandresm** [comment]
#bump

**Pull request merged**
